Makefile

```
@@ -136,6 +136,8 @@ else
   CC_CHECK  = @:
 endif
 
+OBJDUMP_FLAGS := -d -r -z -Mreg-names=32
```

```suggestion
OBJDUMP_FLAGS := --disassemble --reloc --disassemble-zeroes -Mreg-names=32
```
(I don't see a reason to make the makefile more cryptic than it already is 😄)

About `-Mreg-names=32`, could you be so kind as to refresh my memory or link me to a resource about the ABIs. `mips-linux-gnu-objdump --help` lists `numeric 32 n32 64`
Also note there is the less general `gpr-names` and `fpr-names` according to the objdump manual

---

Also I'm not sure on the command line syntax but if `-M` could be `--disassembler-options=` 🙂

---

For others here's a before/after on these changes with `EnDaiku_ChangeAnim` which is a good example (I think) since it has floats and relocs and is somewhat short and I'm a carpenters fan.

before (`objdump -d`):
<details>

```s
00000000 <EnDaiku_ChangeAnim>:
       0:	27bdffc8 	addiu	sp,sp,-56
       4:	afbf0024 	sw	ra,36(sp)
       8:	afa40038 	sw	a0,56(sp)
       c:	afa60040 	sw	a2,64(sp)
      10:	8cc20000 	lw	v0,0(a2)
      14:	3c180000 	lui	t8,0x0
      18:	27180078 	addiu	t8,t8,120
      1c:	04400003 	bltz	v0,2c <EnDaiku_ChangeAnim+0x2c>
      20:	00057900 	sll	t7,a1,0x4
      24:	14a20004 	bne	a1,v0,38 <EnDaiku_ChangeAnim+0x38>
      28:	0005c900 	sll	t9,a1,0x4
      2c:	44800000 	mtc1	zero,$f0
      30:	10000005 	b	48 <EnDaiku_ChangeAnim+0x48>
      34:	01f81821 	addu	v1,t7,t8
      38:	3c080000 	lui	t0,0x0
      3c:	25080078 	addiu	t0,t0,120
      40:	03281821 	addu	v1,t9,t0
      44:	c460000c 	lwc1	$f0,12(v1)
      48:	8c640000 	lw	a0,0(v1)
      4c:	afa3002c 	sw	v1,44(sp)
      50:	afa5003c 	sw	a1,60(sp)
      54:	0c000000 	jal	0 <EnDaiku_ChangeAnim>
      58:	e7a00034 	swc1	$f0,52(sp)
      5c:	44822000 	mtc1	v0,$f4
      60:	8fa3002c 	lw	v1,44(sp)
      64:	c7a00034 	lwc1	$f0,52(sp)
      68:	468021a0 	cvt.s.w	$f6,$f4
      6c:	8fa40038 	lw	a0,56(sp)
      70:	90690008 	lbu	t1,8(v1)
      74:	3c063f80 	lui	a2,0x3f80
      78:	24070000 	li	a3,0
      7c:	8c650000 	lw	a1,0(v1)
      80:	e7a60010 	swc1	$f6,16(sp)
      84:	e7a00018 	swc1	$f0,24(sp)
      88:	2484014c 	addiu	a0,a0,332
      8c:	0c000000 	jal	0 <EnDaiku_ChangeAnim>
      90:	afa90014 	sw	t1,20(sp)
      94:	8faa003c 	lw	t2,60(sp)
      98:	8fab0040 	lw	t3,64(sp)
      9c:	ad6a0000 	sw	t2,0(t3)
      a0:	8fbf0024 	lw	ra,36(sp)
      a4:	27bd0038 	addiu	sp,sp,56
      a8:	03e00008 	jr	ra
      ac:	00000000 	nop
```

</details>

after (`objdump --disassemble --reloc --disassemble-zeroes -Mreg-names=32`)
<details>

```s
00000000 <EnDaiku_ChangeAnim>:
       0:	27bdffc8 	addiu	sp,sp,-56
       4:	afbf0024 	sw	ra,36(sp)
       8:	afa40038 	sw	a0,56(sp)
       c:	afa60040 	sw	a2,64(sp)
      10:	8cc20000 	lw	v0,0(a2)
      14:	3c180000 	lui	t8,0x0
			14: R_MIPS_HI16	.data
      18:	27180078 	addiu	t8,t8,120
			18: R_MIPS_LO16	.data
      1c:	04400003 	bltz	v0,2c <EnDaiku_ChangeAnim+0x2c>
      20:	00057900 	sll	t7,a1,0x4
      24:	14a20004 	bne	a1,v0,38 <EnDaiku_ChangeAnim+0x38>
      28:	0005c900 	sll	t9,a1,0x4
      2c:	44800000 	mtc1	zero,fv0
      30:	10000005 	b	48 <EnDaiku_ChangeAnim+0x48>
      34:	01f81821 	addu	v1,t7,t8
      38:	3c080000 	lui	t0,0x0
			38: R_MIPS_HI16	.data
      3c:	25080078 	addiu	t0,t0,120
			3c: R_MIPS_LO16	.data
      40:	03281821 	addu	v1,t9,t0
      44:	c460000c 	lwc1	fv0,12(v1)
      48:	8c640000 	lw	a0,0(v1)
      4c:	afa3002c 	sw	v1,44(sp)
      50:	afa5003c 	sw	a1,60(sp)
      54:	0c000000 	jal	0 <EnDaiku_ChangeAnim>
			54: R_MIPS_26	Animation_GetLastFrame
      58:	e7a00034 	swc1	fv0,52(sp)
      5c:	44822000 	mtc1	v0,ft0
      60:	8fa3002c 	lw	v1,44(sp)
      64:	c7a00034 	lwc1	fv0,52(sp)
      68:	468021a0 	cvt.s.w	ft1,ft0
      6c:	8fa40038 	lw	a0,56(sp)
      70:	90690008 	lbu	t1,8(v1)
      74:	3c063f80 	lui	a2,0x3f80
      78:	24070000 	li	a3,0
      7c:	8c650000 	lw	a1,0(v1)
      80:	e7a60010 	swc1	ft1,16(sp)
      84:	e7a00018 	swc1	fv0,24(sp)
      88:	2484014c 	addiu	a0,a0,332
      8c:	0c000000 	jal	0 <EnDaiku_ChangeAnim>
			8c: R_MIPS_26	Animation_Change
      90:	afa90014 	sw	t1,20(sp)
      94:	8faa003c 	lw	t2,60(sp)
      98:	8fab0040 	lw	t3,64(sp)
      9c:	ad6a0000 	sw	t2,0(t3)
      a0:	8fbf0024 	lw	ra,36(sp)
      a4:	27bd0038 	addiu	sp,sp,56
      a8:	03e00008 	jr	ra
      ac:	00000000 	nop
```

</details>

---

Looking at what changes, `-Mreg-names=32` turns for example `$f4` into `ft0`. While that's probably correct, don't emulators and debuggers just call it f4? It may make reading these disassemblies a weird experience for debugging something

---

My concern with using the long opts names here is really that you could ask to do that for *everything* in the makefile, which will bloat the output a lot... should we use `--output-target` for objcopy? Or ZAPD? An explanatory comment is probably better.

---

I would think yes, make output is already bloated and you only need it when something goes wrong anyway, at which point I would rather be able to understand the command without refering to the manuals constantly instead of having short commands.

Also if bloat is a concern we could make these objdump calls optional and disable them by default since they aren't involved in the actual build

Alternatively yeah comments would be the bare minimum, though I would like the longer options more

---

We could make the actual commands invisible with `@`.